src/ConstructiveSolidGeometry/VolumePrimitives/Ellipsoid.jl

```diff
@@ -47,13 +47,32 @@ This is a sphere with inner radius 1 and outer radius 2.
 
 See also [Constructive Solid Geometry (CSG)](@ref).
 """
-@with_kw struct Ellipsoid{T,CO,TR,TP,TT} <: AbstractVolumePrimitive{T, CO}
-    r::TR = 1
-    φ::TP = nothing
-    θ::TT = nothing
+struct Ellipsoid{T,CO,TR,TP,TT} <: AbstractVolumePrimitive{T, CO}
+    r::TR 
+    φ::TP 
+    θ::TT 
 
-    origin::CartesianPoint{T} = zero(CartesianPoint{T})
-    rotation::SMatrix{3,3,T,9} = one(SMatrix{3, 3, T, 9})
+    origin::CartesianPoint{T}
+    rotation::SMatrix{3,3,T,9}
+    
+    function Ellipsoid{T,CO}(r::TR,φ::TP,θ::TT,origin::CartesianPoint{T}=zero(CartesianPoint{T}),
+        rotation::SMatrix{3,3,T,9}=one(SMatrix{3, 3, T, 9})) where {T,CO,TR<:T,TP<:Nothing,TT<:Nothing} 
```

Comment:
This `where {T,CO,TR<:T,TP<:Nothing,TT<:Nothing}` is done to limit `TR` to `T`, and `TP` and `TT` to `Nothing`?

Reply:
Correct, at the moment, we only support full spheres